### PR TITLE
Convert StaticAnalysis to .NET Core

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Fix for https://github.com/Azure/azure-powershell/issues/5900

Convert StaticAnalysis project to .NET Core and ensure that the DependencyAnalyzer is run as a part of the .NET Core build.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: tools/StaticAnalysis/BreakingChangeAttributesAnalyzer/BreakingChangeAttributesAnalyzer.cs [hunk 1]
```diff
@@ -226,10 +229,6 @@ private void LogBreakingChangesInModule(BreakingChangeAttributesInModule moduleD
             foreach (BreakingChangeAttributesInCmdlet cmdletData in moduleData.CmdletList)
             {
                 textForBreakingChangesInModule += string.Format(BREAKING_CHANGE_CMDLET_HEADER_FORMAT_STRING, cmdletData.CmdletName);
-                foreach (GenericBreakingChangeAttribute attribute in cmdletData.BreakingChangeAttributes)
```
Comment (cormacpayne) [inline tools/StaticAnalysis/BreakingChangeAttributesAnalyzer/BreakingChangeAttributesAnalyzer.cs]: revert this change

File: tools/StaticAnalysis/BreakingChangeAttributesAnalyzer/CmdletBreakingChangeAttributeLoader.cs [hunk 2]
```diff
@@ -58,14 +56,10 @@ public BreakingChangeAttributesInModule GetModuleBreakingChangeAttributes(string
                 foreach (var type in assembly.GetCmdletTypes())
                 {
                     var cmdlet = type.GetAttribute<CmdletAttribute>();
-                    var attributes = type.GetAttributes<GenericBreakingChangeAttribute>();
```
Comment (cormacpayne) [inline tools/StaticAnalysis/BreakingChangeAttributesAnalyzer/CmdletBreakingChangeAttributeLoader.cs]: revert this change

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -44,7 +44,7 @@ $PSVersionTable
 
 ### Debug Output
 
-<!-- Please run the above script with $DebugPreference = "Continue" and paste the resulting debug stream in the below code block -->
+<!-- Please run the above script with $DebugPreference='Continue' and paste the resulting debug stream in the below code block -->
```
Comment (MiYanni) [inline .github/ISSUE_TEMPLATE.md]: Nice! 👍

File: src/Common/Commands.Common.Storage.Management/Common.Storage.Management.Netcore.csproj [hunk 4]
```diff
@@ -22,7 +22,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Storage" Version="7.1.0-preview" />
```
Comment (MiYanni) [inline src/Common/Commands.Common.Storage.Management/Common.Storage.Management.Netcore.csproj]: Common projects aren't used... why the change(s)?

File: src/Common/Commands.Common.Storage/Common.Storage.Netcore.csproj [hunk 5]
```diff
@@ -26,7 +26,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Storage" Version="7.1.0-preview" />
```
Comment (MiYanni) [inline src/Common/Commands.Common.Storage/Common.Storage.Netcore.csproj]: Common projects aren't used... why the change(s)?

File: src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.Netcore.csproj [hunk 6]
```diff
@@ -30,7 +30,7 @@
   <ItemGroup>
     <PackageReference Include="Azure.Batch" Version="8.1.1" />
     <PackageReference Include="Microsoft.Azure.Management.Batch" Version="4.2.0" />
-    <PackageReference Include="Microsoft.Extensions.Primitives" Version="2.1.1" />
+    <PackageReference Include="Microsoft.Extensions.Primitives" Version="1.1.0" />
```
Comment (MiYanni) [inline src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.Netcore.csproj]: Why the downgrade?
